### ISSUE TITLE
p_id set to '0' when adding new sub-place using previous place

### ISSUE THREAD

**Issue by michelsup** [fisharebest/webtrees#759]
When you edit a place, like the birthplace of a new entry, and search for the few first letter of a previously edited place, you get the field populated, e.g "St Céré, Lot, Midi-Pyrénées, France".
If you add another sub-place in front of this, like "Nowhere, St Céré, Lot, Midi-Pyrénées, France", you end up, in your database, with an entry like 
p_id | p_place | p_parent_id | p_file | p_std_soundex | p_dm_soundex
0      | Nowhere | 129            | 1        |L000:V253:L125 | 800000:756349:746349:875634:874634

This p_id  screws up everything, as you can imagine. I don't have time to check the code involved, but I can, if you need, provide you with logs and such.

**Comment by fisharebest** [comment]
1) by default, if you insert zero into an auto-increment field, it is set to the next value.

Have you set your SQL_MODE to include NO_AUTO_VALUE_ON_ZERO?

2) We never actually insert zero.  We insert NULL.  More specifically, we insert all the other columns, and allow MySQL to populate the field.

So it should not be possible for webtrees to create a row with zero in this column

Can you provide the output from `SHOW CREATE TABLE wt_places;`

**Comment by michelsup** [comment]
Thank you for your quick reply.

SHOW CREATE TABLE wt_places;

CREATE TABLE `wt__places` (
 `p_id` int(11) NOT NULL DEFAULT '0',
 `p_place` varchar(150) COLLATE utf8_unicode_ci DEFAULT NULL,
 `p_parent_id` int(11) DEFAULT NULL,
 `p_file` int(11) NOT NULL,
 `p_std_soundex` text COLLATE utf8_unicode_ci,
 `p_dm_soundex` text COLLATE utf8_unicode_ci
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci

Obviously, as you guessed, p_id is not set properly, as it's not incremented.
I have not set any SQL_MODE anywhere and I have not tweaked anything, this is a plain vanilla installation.
I change p_id to be A.I, let's see.

**Comment by fisharebest** [comment]
> p_id int(11) NOT NULL DEFAULT '0',

The script to create this table uses `p_id INTEGER AUTO_INCREMENT NOT NULL`.
See `app\schema\Migration0.php`.

I can't see how your table definition could be different.  Have you done anything that migh change it?

**Comment by michelsup** [comment]
I just installed webtrees a while ago and imported my gedcom from phpgedview. I never ever tweaked anything within webtrees or setup anything strange within mysql. As I said, plain vanilla, and it's the most worrying. 
So, I altered as Migration0.php, but obviously I can't make unique p_parent_id, p_file, p_place.
I noticed that wt_placelinks is fucked up aswell, as I have no primary keys and all my pl_p_id and pl_file are "1".
Big , BIG problem.
